shor/gates.py

```diff
@@ -489,14 +1388,68 @@ def to_matrix(self) -> np.ndarray:
 
 
 class U2(U3):
+    """
+    apply U2 gate for single qubit rotation with 2 euler angles
```

Comment:
Note U2 and U1 are both getting deprecated, and U3 is becoming U on qiskit.

Reply:
Good to know thanks, We can make this change in a future issue